cmd/utreexoserver/bridgeserver.go

```diff
@@ -114,13 +117,22 @@ func main() {
 	sig := make(chan bool, 1)
 	handleIntSig(sig, *cpuProfCmd, *traceCmd)
 
-	fmt.Printf("datadir is %s\n", dataDir)
-	err := bridge.BuildProofs(param, dataDir, *forestInRam, *forestCache, sig)
+	// only do buildProofs or serve; need to restart to serve after
+	// building proofs
+
+	if !*serve {
+		fmt.Printf("datadir is %s\n", dataDir)
+		err := bridge.BuildProofs(param, dataDir, *forestInRam, *forestCache, sig)
+		if err != nil {
+			fmt.Printf("Buildproofs error: %s\n", err.Error())
+			panic("proof build halting")
+		}
+	}
+	err := bridge.ArchiveServer(param, dataDir, sig)
 	if err != nil {
-		fmt.Printf("Buildproofs error: %s\n", err.Error())
+		fmt.Printf("ServeBlocks error: %s\n", err.Error())
```

Comment:
where does the name "ServeBlocks" come from? wouldn't it be better to call it "ArchiveServer error" since that is the function that errored?

Reply:
Ah yup, missed that! thanks